zookeeper/core/component_test.py

```diff
@@ -787,3 +787,21 @@ class Parent:
 
     assert p.grand_child.a == 3
     assert p.child.grand_child.a == 3
+
+
+def test_component_configure_component_passed_as_config():
+    @component
+    class Child:
+        x: int = Field()  # Inherited from parent
+
+    @component
+    class Parent:
+        x: int = Field(7)
+        child: Child = ComponentField(Child)
+
+    instance = Parent()
+    new_child_instance = Child()
+    configure(instance, {"child": new_child_instance})
+    assert instance.child is new_child_instance
+    assert instance.child.__component_parent__ is instance
+    assert instance.child.x == 7  # This value should be correctly inherited.
```

Comment:
I've verified that this test works, in the sense that it fails if I remove the change in `component.py`.